src/loggers.ts

```diff
@@ -0,0 +1,95 @@
+import { CancelError, HTTPError, Options, Response } from 'got';
+
+export interface Logger {
+  info(...args: unknown[]): void;
+  error(...args: unknown[]): void;
+  debug(...args: unknown[]): void;
+}
+
+export interface HttpLogger {
+  logSuccess(response: Response, options: Options): void;
+  logFailure(error: HTTPError | CancelError): void;
+}
+
+export class NoOpLogger implements HttpLogger {
+  // eslint-disable-next-line class-methods-use-this
+  logSuccess(_response: Response, _options: Options): void {}
+
+  // eslint-disable-next-line class-methods-use-this
+  logFailure(_error: HTTPError | CancelError): void {}
+}
+
+export class GenericLogger implements HttpLogger {
+  readonly #logger: Logger;
+
+  constructor(logger: Logger) {
+    this.#logger = logger;
+  }
+
+  logSuccess(response: Response, options: Options): void {
+    const { url, statusCode, timings } = response;
+    const message = `${options.method} ${url} ${statusCode} ${new Date().getTime() - timings.start} ms`;
+    if (options.method === 'GET') {
+      this.#logger.debug(message);
+    } else {
+      this.#logger.info(message);
+    }
```

Comment:
I'm honestly not a huge fan of this opinionation here, but not sure how to filter it on the application side in a better way so as not to emit a log statement at all.